### PR TITLE
Refactor renderer::Camera and fix warnings on Windows

### PR DESCRIPTION
**Pull request by dictoon**

### CODE REVIEW

File: src/appleseed/renderer/modeling/camera/camera.cpp [hunk 1]
```diff
@@ -64,6 +64,40 @@ namespace
...
+
+    // Normalization factor to make the area under the curve equal to 1.
+    float                       m_shutter_curve_bezier_normalization_factor;
+
+    // Polynomials of open/close CDFs.
+    Vector<float, 7>            m_shutter_curve_bezier_open_cdf;
+    Vector<float, 7>            m_shutter_curve_bezier_close_cdf;
+
+    Impl()
+      : m_use_bezier_shutter_curve(false)
```
Comment (fedormatantsev) [inline src/appleseed/renderer/modeling/camera/camera.cpp]: This probably should go to on_render_begin()

File: src/appleseed/renderer/modeling/camera/camera.h [hunk 2]
```diff
@@ -171,26 +177,6 @@ class APPLESEED_DLLSYMBOL Camera
     float               m_inverse_cdf_close_point;
```
Comment (fedormatantsev) [inline src/appleseed/renderer/modeling/camera/camera.h]: I suggest to move inverse_cdf_close_point/inverse_cdf_open_point to impl as well, those two are internal vars used only for inverse sampling. Normalized times seem to be internal vars as well.

File: src/appleseed/renderer/modeling/camera/camera.cpp (hunk 1 above)
Comment (dictoon) [reply src/appleseed/renderer/modeling/camera/camera.cpp]: Good point!

File: src/appleseed/renderer/modeling/camera/camera.h (hunk 2 above)
Comment (dictoon) [reply src/appleseed/renderer/modeling/camera/camera.h]: Yes, makes sense.

**Comment by dictoon** [comment]
Thanks for the review, pushing a new commit in a minute (I'll squash at the end).

File: src/appleseed/renderer/modeling/camera/camera.h [hunk 3]
```diff
@@ -173,8 +173,6 @@ class APPLESEED_DLLSYMBOL Camera
     float               m_shutter_open_time_interval;
     float               m_normalized_open_end_time;
     float               m_normalized_close_start_time;
```
Comment (fedormatantsev) [inline src/appleseed/renderer/modeling/camera/camera.h]: I also mentioned normalized times, they are implementation specific too. Sorry if that wasn't clear.

Comment (dictoon) [reply src/appleseed/renderer/modeling/camera/camera.h]: Hadn't read your message until the end :) Pushing a new commit with a few renamings as well in an attempt to reduce verbosity somewhat.

**Comment by dictoon** [comment]
Cool, thanks!

There are two final renamings I want to do:
- `m_shutter_open_time` -> `m_shutter_open_start_time`
- `m_shutter_close_time` -> `m_shutter_close_end_time`

I'll also replace "start" by "begin" in order to be consistent with "end".

**Comment by fedormatantsev** [comment]
Sounds good. This will make naming a lot more consistent.

**Pull request merged**